### PR TITLE
80: Added button to open modal to show server startup command

### PR DESCRIPTION
**Pull request by Ivan0348**

### CODE REVIEW

File: frontend/src/pages/server/Administration.tsx [hunk 1]
```diff
@@ -127,9 +127,49 @@ export function Administration({setServ, serv, onServerFilesDeleted}: Props) {
...
+                                        <DialogActions>
+                                            <Button variant="solid" color="neutral" onClick={() => setShowServerCommandModalOpen(false)}>
+                                                Close
+                                            </Button>
+                                        </DialogActions>
+                                    </ModalDialog>
+                                </Modal>
+                                <Button color='neutral' onClick={() => {
+                                    setShowServerCommandModalOpen(true)
+                                    GetServerWithError(serv.id, false).then(() => {
```
Comment (JensvandeWiel) [inline frontend/src/pages/server/Administration.tsx:156]: you are not actually using the retunred server, so you can remove GetServerWithError

File: server/server.go [hunk 2]
```diff
@@ -79,10 +79,7 @@ func (s *Server) Start() error {
 	if s.IsServerRunning() {
 		return fmt.Errorf("error starting server: server is already running")
 	} else {
-
-		args := s.CreateArguments()
-
-		s.Command = exec.Command(path.Join(s.ServerPath, "ShooterGame\\Binaries\\Win64\\ArkAscendedServer.exe"), args...)
+		s.Command = s.GetServerCommand()
```
Comment (JensvandeWiel) [inline server/server.go:82]: Rename to CreateServerCmd

File: server/server_controller.go [hunk 3]
```diff
@@ -493,3 +493,11 @@ func (c *ServerController) CheckServerInstalled(id int) (bool, error) {
 }
 
 //endregion
+
+func (c *ServerController) GetServerCommandWrapper(id int) string {
```
Comment (JensvandeWiel) [inline server/server_controller.go:497]: rename to GetServerStartupCommand

File: server/server_controller.go [hunk 4]
```diff
@@ -493,3 +493,11 @@ func (c *ServerController) CheckServerInstalled(id int) (bool, error) {
...
+	server, err := c.GetServerWithError(id, false)
+	if err != nil {
+		return "error getting server command: " + err.Error()
```
Comment (JensvandeWiel) [inline server/server_controller.go:500]: Handle error like other functions

**Review by JensvandeWiel — CHANGES_REQUESTED** [review]

**Comment by Ivan0348** [comment]
Can you review again?

File: server/server_controller.go
Comment (JensvandeWiel) [inline server/server_controller.go:1]: also return the error to the frontend like (string, error)
look at this for example:
```go
func (c *ServerController) GetServerStatus(id int) (bool, error) {
	server, exists := c.Servers[id]
	if !exists {
		err := fmt.Errorf("error getting server status " + strconv.Itoa(id) + ": server does not exist in map")
		runtime.LogError(c.ctx, err.Error())
		return false, err
	}

	status := server.IsServerRunning()
	return status, nil
}
```

File: frontend/src/pages/server/Administration.tsx [hunk 5]
```diff
@@ -127,9 +127,47 @@ export function Administration({setServ, serv, onServerFilesDeleted}: Props) {
...
+                                </Modal>
+                                <Button color='neutral' onClick={() => {
+                                    setShowServerCommandModalOpen(true)
+                                        return GetServerStartupCommand(serv.id)
+                                            .then((cmd: string) => {
+                                            setServerCommand(cmd)
+                                            }).catch((err) => {
+                                                console.error(err);
+                                                addAlert(err, "danger");
+                                            }).catch((err: string) => {
```
Comment (JensvandeWiel) [inline frontend/src/pages/server/Administration.tsx:162]: you have a double .catch you can remove one

File: server/server_controller.go [hunk 6]
```diff
@@ -493,3 +493,11 @@ func (c *ServerController) CheckServerInstalled(id int) (bool, error) {
 }
 
 //endregion
+
+func (c *ServerController) GetServerStartupCommand(id int) string {
```
Comment (JensvandeWiel) [inline server/server_controller.go:497]: also return the error to the frontend like (string, error)
look at this for example:
```go
func (c *ServerController) GetServerStatus(id int) (bool, error) {
	server, exists := c.Servers[id]
	if !exists {
		err := fmt.Errorf("error getting server status " + strconv.Itoa(id) + ": server does not exist in map")
		runtime.LogError(c.ctx, err.Error())
		return false, err
	}

	status := server.IsServerRunning()
	return status, nil
}
```

**Review by JensvandeWiel — CHANGES_REQUESTED** [review]

**Comment by Ivan0348** [comment]
can you review again?

**Review by JensvandeWiel — APPROVED** [review]
Great job!

**Pull request merged**